tasmota/xnrg_15_teleinfo.ino

```diff
@@ -0,0 +1,422 @@
+{
+    char c = ' ';
+
+    // Does this value is new or changed?
+    if (flags & (TINFO_FLAGS_ADDED | TINFO_FLAGS_UPDATED) ) 
+    {
+        if (flags & TINFO_FLAGS_ADDED)   { c = '#';  }
+        if (flags & TINFO_FLAGS_UPDATED) { c = '*';  }
+
+        // Current tarif
+        if (!strcmp_P(LABEL_PTEC, me->name))
+        {
+            if (!strcmp_P("TH..", me->name)) { tarif = TARIF_TH; }
+            else if (!strcmp_P("HC..", me->name)) { tarif = TARIF_HC; }
+            else if (!strcmp_P("HP..", me->name)) { tarif = TARIF_HP; }
+            else if (!strcmp_P("HN..", me->name)) { tarif = TARIF_HN; }
+            else if (!strcmp_P("PM..", me->name)) { tarif = TARIF_PM; }
+            else if (!strcmp_P("HCJB", me->name)) { tarif = TARIF_CB; }
+            else if (!strcmp_P("HCJW", me->name)) { tarif = TARIF_CW; }
+            else if (!strcmp_P("HCJR", me->name)) { tarif = TARIF_CR; }
+            else if (!strcmp_P("HPJB", me->name)) { tarif = TARIF_PB; }
+            else if (!strcmp_P("HPJW", me->name)) { tarif = TARIF_PW; }
+            else if (!strcmp_P("HPJR", me->name)) { tarif = TARIF_PR; }
+
+            AddLog_P2(LOG_LEVEL_DEBUG, PSTR("TIC: Tarif changed, now '%s' (%d)"), me->value, tarif);
+        } 
+        // Voltage V (not present on all Smart Meter)
+        else if (!strcmp_P(LABEL_TENSION, me->name))
+        {
+             Energy.voltage_available = true;
+             int i = atoi(me->value);
+             Energy.voltage[0]  = (float) atoi(me->value);
+
+            // Update current
+            if (Energy.voltage_available && Energy.voltage[0]) {
+                Energy.current[0] = Energy.active_power[0] / Energy.voltage[0] ;
+            }
+
+             AddLog_P2(LOG_LEVEL_DEBUG, PSTR("TIC: Voltage %s, now %d"), me->value, i);
+        }
+        // Current I
+        else if (!strcmp_P(LABEL_IINST, me->name))
+        {
+             if (!Energy.voltage_available) {
+                int i = atoi(me->value);
+                Energy.current[0]  = (float) atoi(me->value);
+             } else if (Energy.voltage[0]) {
+                Energy.current[0] = Energy.active_power[0] / Energy.voltage[0] ;
+             }
+
+             AddLog_P2(LOG_LEVEL_DEBUG, PSTR("TIC: Current %s, now %d"), me->value, (int) Energy.current[0]);
+        }
+        // Current P
+        else if (!strcmp_P(LABEL_PAPP, me->name))
+        {
+             int papp = atoi(me->value);
+             Energy.active_power[0] = (float) atoi(me->value);
+             AddLog_P2(LOG_LEVEL_DEBUG, PSTR("TIC: Power %s, now %d"), me->value, papp);
+
+            // Update current
+            if (Energy.voltage_available && Energy.voltage[0]) {
+                Energy.current[0] = Energy.active_power[0] / Energy.voltage[0] ;
+            }
+        }
+        // kWh indexes
+        else if (!strcmp_P(LABEL_HCHC, me->name) || !strcmp(LABEL_HCHP, me->name))
+        {
+            char value[32];
+            unsigned long hc = 0;
```

Comment:
Got it, index should fit in uint_32, can I schedule this one on next PR because currently my file had a lot of change that I can't PR right now, such as waiting also PR #8701